test/amqp/test_publish.py

```diff
@@ -150,7 +149,7 @@ def test_confirms(self, amqp_uri, publisher):
 
         # confirms are enabled by default;
         # if a mandatory message cannot be delivered, expect an exception
-        with pytest.raises(UndeliverableMessage):
+        with pytest.raises(ChannelError):
```

Comment:
Review comment got lost somehow:

I think we need to keep `UndeliverableMessage` when `publisher.publish` can't deliver a mandatory message.